sdks/python/setup.py

```diff
@@ -289,7 +289,8 @@ def get_portability_package_data():
             'google-cloud-bigquery-storage>=2.6.3,<2.14',
             'google-cloud-core>=0.28.1,<3',
             'google-cloud-bigtable>=0.31.1,<2',
-            'google-cloud-spanner>=1.13.0,<2',
+            # google-cloud-spanner 2.x causes dependency parsing backoff
+            'google-cloud-spanner>=1.13.0,!=2,<=3.21.0',
```

Comment:
1. We shouldn't use <= in the upper bound, it precludes us from asking someone to issue a patch release that could still be picked up by the constraints.
So we can use `<3.22.0` or `<4`. 

2. I think this is server-side dependency only, so using `<4` should be safe as we do with other GCP dependencies at this time. We'll decide soon whether we want to cap upper bounds for all IO.

3. Can you please run PostCommit integration tests with `google-cloud-spanner>=3.0.0` requirement if you haven't ? Unless we specify that requirement, postcommit tests will not pick up the new dependency.

Reply:
(1,2) Yes we should set 'google-cloud-spanner<4' as other cloud dependencies

(3) I ran this when debugging (got some error and fixed it). From [logging](https://ci-beam.apache.org/job/beam_PostCommit_Python37_PR/441/consoleFull) it shows version 3.21.0 is picked by pypi:
```
15:20:37 Collecting google-cloud-spanner!=2,<=3.21.0,>=1.13.0
15:20:37   Using cached google_cloud_spanner-3.21.0-py2.py3-none-any.whl (291 kB)
```
also
```
installed: apache-beam @ file:///app/sdks/python/target/.tox/.tmp/package/1/apache-beam-2.43.0.dev0.zip,cachetools==4.2.4,...,google-cloud-recommendations-ai==0.7.1,google-cloud-spanner==3.21.0,...
```
jenkins already cached v3.21.0 and is using it.

Reply:
Thanks, change applied.